modules/programs/k9s.nix

```diff
@@ -140,31 +163,67 @@ in {
-      source = yamlFormat.generate "k9s-config" cfg.settings;
-    };
-
-    xdg.configFile."k9s/skin.yml" = mkIf (cfg.skin != { }) {
-      source = yamlFormat.generate "k9s-skin" cfg.skin;
-    };
-
-    xdg.configFile."k9s/aliases.yml" = mkIf (cfg.aliases != { }) {
-      source = yamlFormat.generate "k9s-aliases" cfg.aliases;
-    };
-
-    xdg.configFile."k9s/hotkey.yml" = mkIf (cfg.hotkey != { }) {
-      source = yamlFormat.generate "k9s-hotkey" cfg.hotkey;
-    };
-
-    xdg.configFile."k9s/plugin.yml" = mkIf (cfg.plugin != { }) {
-      source = yamlFormat.generate "k9s-plugin" cfg.plugin;
+  config = let
+    isAtLeast0_29 = (lib.versionAtLeast cfg.package.version "0.29.0");
+    isAtLeast0_30 = (lib.versionAtLeast cfg.package.version "0.30.0");
+    fileExt = if isAtLeast0_30 then "yaml" else "yml";
+    skinName = cfg.settings.k9s.ui.skin or (if cfg.skins != { } then
+      "${builtins.elemAt (builtins.attrNames cfg.skins) 0}"
+    else
+      "skin");
+
+    skinSetting = if (isAtLeast0_29 && !cfg.settings ? k9s.ui.skin) then {
+      k9s.ui.skin = skinName;
+    } else
+      { };
+    skinFiles = mapAttrs' (name: value:
+      nameValuePair "k9s/skins/${name}.${fileExt}" {
+        source = yamlFormat.generate "k9s-skin-${name}.${fileExt}" value;
+      }) cfg.skins;
+
+    # TODO: remove after suitable deprecation period
+    deprecatedSkinOpt = let
+      skinPath = if isAtLeast0_29 then
+        "skins/${skinName}.${fileExt}"
+      else
+        "skin.${fileExt}";
+    in {
+      "k9s/${skinPath}" =
+        mkIf (cfg.skin != { } || ((cfg.skins != { }) && !isAtLeast0_29)) {
+          source = yamlFormat.generate "k9s-skin"
+            (if cfg.skin != { } then cfg.skin else cfg.skins."${skinName}");
+        };
     };
+  in mkIf cfg.enable {
+    home.packages = [ cfg.package ];
 
-    xdg.configFile."k9s/views.yml" = mkIf (cfg.views != { }) {
-      source = yamlFormat.generate "k9s-views" cfg.views;
-    };
+    xdg.configFile = {
+      "k9s/config.${fileExt}" = mkIf (cfg.settings != { }) {
+        source = yamlFormat.generate "k9s-config"
+          (lib.recursiveUpdate skinSetting cfg.settings);
+      };
+
+      "k9s/aliases.${fileExt}" = mkIf (cfg.aliases != { }) {
+        source = yamlFormat.generate "k9s-aliases" cfg.aliases;
+      };
+
+      "k9s/hotkey.${fileExt}" = mkIf (cfg.hotkey != { }) {
+        source = yamlFormat.generate "k9s-hotkey" cfg.hotkey;
+      };
+
+      "k9s/plugin.${fileExt}" = mkIf (cfg.plugin != { }) {
+        source = yamlFormat.generate "k9s-plugin" cfg.plugin;
+      };
+
+      "k9s/views.${fileExt}" = mkIf (cfg.views != { }) {
+        source = yamlFormat.generate "k9s-views" cfg.views;
+      };
+    } // deprecatedSkinOpt // skinFiles;
+
+    warnings = optional (cfg.skin != { }) ''
+      Using programs.k9s.skin is deprecated and will be
+      removed in the future. Please change to using programs.k9s.skins
+      as an attribute set of one or more skin definitions.
+    '';
```

Comment:
Perhaps it would be sufficient to use, `mkRenamedOptionModule`? Something like

```nix
imports = [
  (mkRenamedOptionModule [ "programs" "k9s" "skin" ] [ "programs" "k9s" "skins" "skin"])
];
```

perhaps? It goes between the `meta.maintainers` and `options.programs.k9s` lines.

Reply:
Potentially. The new `skins` option does expect a slightly different structure, though.
Old `skin`: YAML file content
New `skins`: attrset of { fileName = { YAML file content }, fileName2 = { YAML file content } }

Reply:
It worked well! Thanks for the elegant solution.